File: src/sort/select_sort.php
<?php

$array = [4, 3, 2, 6, 1];
print_r($array);
$result = selectSort($array);
print_r($result);

function selectSort(array $array)
{
    $length = count($array);
    if ($length < 2) return $array;

    for ($i = 0; $i < $length; $i++) {
        $minIndex = $i;
        for ($j = $i + 1; $j < $length; $j++) {
            if ($array[$minIndex] > $array[$j]) {
                $minIndex = $j;
            }
        }
        swap($array, $i, $minIndex);
    }
    return $array;
}

function swap(&$array, $i, $j)
{
    $tmp = $array[$i];
    $array[$i] = $array[$j];
    $array[$j] = $tmp;
}

File: src/sort/select_sort.go
package main

import (
	"fmt"
)

func main() {
	arr := []int{4, 3, 2, 6, 1}
	fmt.Print(arr)
	result := selectSort(arr)
	fmt.Println(result)
}

func selectSort(arr []int) []int {
	if len(arr) < 2 {
		return arr
	}
	for i := 0; i < len(arr); i++ {
		minIndex := i
		for j := i + 1; j < len(arr); j++ {
			if arr[minIndex] > arr[j] {
				minIndex = j
			}
		}
		swap(arr, i, minIndex)
	}
	return arr
}

func swap(arr []int, i int, j int) {
	tmp := arr[i]
	arr[i] = arr[j]
	arr[j] = tmp
}
